MetadataMigration/src/test/java/org/opensearch/migrations/EndToEndTest.java

```diff
@@ -44,13 +44,13 @@ class EndToEndTest extends BaseMigrationTest {
 
     private static Stream<Arguments> scenarios() {
         return SupportedClusters.supportedPairs(false).stream()
-            .filter(pair -> !VersionMatchers.isES_8_X.test(pair.source().getVersion()))
             .flatMap(pair -> {
                 List<TemplateType> templateTypes = Stream.concat(
                             (VersionMatchers.isOS_2_X.test(pair.source().getVersion())
                                     ? Stream.empty()
                                     : Stream.of(TemplateType.Legacy)),
                             (VersionMatchers.equalOrGreaterThanES_7_10.test(pair.source().getVersion())
+                                    || VersionMatchers.isES_8_X.test(pair.source().getVersion())
```

Comment:
I'm not sure what this means at quick inspection or why ES8 should be treated differently, can you combine predicates using `.or(...)`?